dags/utils/slack_operator.py

```diff
@@ -15,74 +18,125 @@ def get_central_time_exec_data(context):
-            *Log Url*: {log_url} 
-            """.format(
-        task=context.get("task_instance").task_id,
-        dag=context.get("task_instance").dag_id,
-        ti=context.get("task_instance"),
-        exec_date=get_central_time_exec_data(context),
-        log_url=context.get("task_instance").log_url,
+
+    task_instance = context.get("task_instance")
+    task = context.get("task")
+    exception = context.get("exception")
+    exception_type = type(exception).__name__ if exception else "Unknown"
+    exception_message = (
+        str(exception) if exception else "No exception message available"
     )
+
+    # Extract additional information
+    dag = context.get("dag")
+    dag_id = task_instance.dag_id
+    task_id = task_instance.task_id
+    exec_date = get_central_time_exec_data(context)
+    log_url = task_instance.log_url
+    try_number = task_instance.try_number
+    max_tries = task.retries  # Accessed from the task
+    operator = task.__class__.__name__  # Gets the operator class name
+    duration = getattr(task_instance, "duration", "Not available")
+
+    schedule_interval = dag.schedule_interval if dag else None
+
+    # Convert schedule_interval to human-readable format
+    if schedule_interval is None:
+        schedule_description = "None"
+    elif isinstance(schedule_interval, str):
+        try:
+            schedule_description = get_description(schedule_interval)
+        except Exception:
+            schedule_description = f"Cron expression: {schedule_interval}"
+    elif isinstance(schedule_interval, datetime.timedelta):
+        # Format timedelta to human-readable string
+        total_seconds = int(schedule_interval.total_seconds())
+        periods = [
+            ("day", 86400),  # 60 * 60 * 24
+            ("hour", 3600),  # 60 * 60
+            ("minute", 60),
+            ("second", 1),
+        ]
+        parts = []
+        for period_name, period_seconds in periods:
+            if total_seconds >= period_seconds:
+                period_value, total_seconds = divmod(total_seconds, period_seconds)
+                part = f"{period_value} {period_name}{'s' if period_value > 1 else ''}"
+                parts.append(part)
+        schedule_description = "Every " + ", ".join(parts)
+    else:
+        schedule_description = str(schedule_interval)
+
+    if (
+        max_tries == 0
+    ):  # if no retry number is allowed, this returns 0, so make it "1" to include the implied, non-repeating try
+        max_tries = 1
+
+    importance = getattr(dag, "importance", None)
+
+    slack_msg = f"""
+:red_circle: *Task Failure Alert*
+{importance}
+
+*DAG*: `{dag_id}`
+*Task*: `{task_id}`
+*Execution Time*: `{exec_date}`
+*Schedule*: `{schedule_description}`
+*Try Number*: `{try_number} of {max_tries}`
+*Duration*: `{duration} seconds`
+*Operator*: `{operator}`
+*Exception Type*: `{exception_type}`
+*Exception Message*: `{exception_message}`
+<{log_url}|*Log URL*>
+"""
+
```

Comment:
I thought I'd take a stab in the dark at what an enhanced notification could look like. I used the o1 model to crank out some overly verbose boilerplate and trimmed and modified it to what I think we've been envisioning.

I have no attachment to any of this, but I thought it was interesting enough to bring up for PR review. If anyone wants to modify it, please feel free to commit on this branch, 💯